packages/insomnia/src/ui/components/base/button.tsx

```diff
@@ -17,34 +12,39 @@ export interface ButtonProps<T> {
   style?: CSSProperties;
 }
 
-@autoBindMethodsForReact(AUTOBIND_CFG)
-export class Button<T> extends PureComponent<ButtonProps<T>> {
-  _handleClick(event: React.MouseEvent<HTMLButtonElement>) {
-    const { onClick, onDisabledClick, disabled, value } = this.props;
-    const fn = disabled ? onDisabledClick : onClick;
+export const Button = <T, >(props: ButtonProps<T>) => {
+  // Distract all the available properties.
+  const {
+    value,
+    children,
+    disabled,
+    tabIndex,
+    className,
+    type,
+    id,
+    style,
+    title,
+    onClick,
+  } = props;
```

Comment:
totally acceptable to destructure these inside the first argument (as long as you don't need the full props object anywhere, and this seems to be one of those places
```suggestion
export const Button = <T, >({
    value,
    disabled,
    tabIndex,
    className,
    type,
    id,
    style,
    title,
    onClick,
  }: PropsWithChildren<ButtonProps<T>>) => {
```

Also, this is a good place to use PropsWithChildren.  Check out `EnumSetting` or `NoticeRow` for examples.

Reply:
@dimitropoulos You are right, many thanks :)